ios/FluentUI.Demo/FluentUI.Demo/Demos/CommandBarDemoController.swift

```diff
@@ -0,0 +1,192 @@
+        super.viewDidLoad()
+        container.layoutMargins.right = 0
+        container.layoutMargins.left = 0
+        view.backgroundColor = Colors.surfaceSecondary
+
+        let itemGroups: [CommandBarItemGroup] = [
+            [
+                Item(command: .add),
+                Item(command: .mention, isEnabled: false),
+                Item(command: .calendar)
+            ],
+            [
+                Item(command: .textBold),
+                Item(command: .textItalic),
+                Item(command: .textUnderline),
+                Item(command: .textStrikethrough)
+            ],
+            [
+                Item(command: .arrowUndo),
+                Item(command: .arrowRedo)
+            ],
+            [
+                Item(command: .copy),
+                Item(command: .delete)
+            ],
+            [
+                Item(command: .checklist),
+                Item(command: .bulletList),
+                Item(command: .numberList),
+                Item(command: .link)
+            ]
+        ]
+
+        if #available(iOS 14.0, *) {
+            // Copy item
+            let copyItem = itemGroups[3][0]
+            copyItem.menu = UIMenu(children: [UIAction(title: "Copy Image", image: UIImage(named: "copy24Regular"), handler: { _ in }),
+                                              UIAction(title: "Copy Text", image: UIImage(named: "text24Regular"), handler: { _ in })])
+            copyItem.showsMenuAsPrimaryAction = true
+        }
+
+        container.addArrangedSubview(createLabelWithText("Default"))
+
+        let defaultCommandBar = CommandBar(itemGroups: itemGroups)
+        defaultCommandBar.backgroundColor = Colors.Navigation.System.background
+        defaultCommandBar.delegate = self
+        container.addArrangedSubview(defaultCommandBar)
+
+        container.addArrangedSubview(createLabelWithText("With Fixed Button"))
+
+        let fixedButtonCommandBar = CommandBar(itemGroups: itemGroups, trailingItem: Item(command: .keyboard))
+        fixedButtonCommandBar.backgroundColor = Colors.Navigation.System.background
+        fixedButtonCommandBar.delegate = self
+        container.addArrangedSubview(fixedButtonCommandBar)
+
+        container.addArrangedSubview(createLabelWithText("In Input Accessory View"))
+
+        let accessoryCommandBar = CommandBar(itemGroups: itemGroups, trailingItem: Item(command: .keyboard))
+        accessoryCommandBar.delegate = self
+        textField.inputAccessoryView = accessoryCommandBar
+        container.addArrangedSubview(textField)
+    }
+
+    func createLabelWithText(_ text: String = "") -> Label {
+        let label = Label(style: .subhead, colorStyle: .regular)
+        label.text = text
+        label.textAlignment = .center
+        return label
+    }
+}
+
+extension CommandBarDemoController: CommandBarDelegate {
+    func commandBar(_ commandBar: CommandBar, didSelectItem item: CommandBarItem) {
+        guard let item = item as? Item else {
```

Comment:
A solution would be adding a block property to `CommandBarItem` instead of a delegate method to handle item selection, but this would not be aligned with designs in other Fluent UI views. What do you think?

Reply:
@rdeassis what is the Fluent preference here? Completion blocks seem the right design in terms of Apple platforms, but it may not be preferred for Fluent?

Reply:
Although there are no official standards defined in FluentUI, I believe the order of preference to be:

1. Completion blocks/closures
2. Protocol/delegate pattern
3. Selectors

Reply:
That seems like the right option then, @zhuorantan would you be able to incorporate a completion block for the item?